src/components/navigation.tsx

```diff
@@ -0,0 +1,100 @@
+'use client';
+import {Disclosure} from '@headlessui/react';
```

Comment:
Very minor: add a newline after line 1, to make visually clear this is a client component (i.e. to separate it from the imports)?